msrestazure/azure_active_directory.py

```diff
@@ -559,11 +559,23 @@ def signed_session(self):
         return session
 
 
-def get_msi_token(resource, port=50342):
+def get_msi_token(resource, port=50342, msi_conf=None):
+    """Get MSI token from inside a VM/VMSS.
+
+    If msi_conf is used, must be a dict of one key in ["client_id", "object_id", "msi_res_id"]
+
+    :param str resource: The resource where the token would be use.
+    :param int port: The port is not the default 50342 is used.
+    :param dict[str, str] msi_conf: msi_conf if User Assigned (if not specified, assume System Assigned)
```

Comment:
Suggest wording of `if to request a token through a user assigned identity`. But up to you